### PR TITLE
feat: new OnchainName component

### PR DESCRIPTION
**Pull request by alvaroraminelli**
**What changed? Why?**

Adding the first `OnchainName` component of our  Identity Kit 👤

The `OnchainName` component primarily focuses on showcasing ENS names for given Ethereum addresses, and defaults to displaying a sliced version of the address when an ENS name isn't available.

```tsx
import { OnchainName } from '@coinbase/onchainkit';

<OnchainName address="0x1234567890abcdef1234567890abcdef12345678" sliced={false} />;
```

**Notes to reviewers**

**How has it been tested?**

- Unit tests only.

### CODE REVIEW

File: README.md
Comment (Zizzamia) [inline README.md:216]: This new section will be called Identity Kit, which will have a callection of components and utilities around Onchain Identity.

File: README.md
Comment (Zizzamia) [inline README.md:220]: How about
```
The `OnchainName` component is used to display ENS names associated with Ethereum addresses. When an ENS name is not available, it defaults to showing a truncated version of the address.
```

File: README.md
Comment (Zizzamia) [inline README.md:228]: Oh, I think we can use types going forward to better describe what's going on. Check the latest README.

File: README.md
Comment (Zizzamia) [inline README.md:225]: I think we can do a bit more here to showcase a real example, like have a small component using `<OnchainName>`

And have as comment how the address will result.

File: src/components/OnchainName.tsx
Comment (Zizzamia) [inline src/components/OnchainName.tsx:7]: I think the `address` should be mandatory

File: src/components/OnchainName.tsx
Comment (Zizzamia) [inline src/components/OnchainName.tsx:26]: Add a comment here why we have `useMemo`

Comment (Zizzamia) [reply src/components/OnchainName.tsx:26]: Also should we avoid have CPU execution for the `normilizedAddress` if in case of ensName we will use that anyway?

File: src/store/client.ts
Comment (Zizzamia) [inline src/store/client.ts:4]: I think `client` should go in `src/network` or more a location where we handle Data Layer stuff. cc @robpolak

File: src/store/inMemoryStorageService.ts
Comment (Zizzamia) [inline src/store/inMemoryStorageService.ts:5]: Let's convert this to functional at some point.

**Comment by Zizzamia** [comment]
Before we merge, let's have also a quick review from @cnasc, @robpolak, @wespickett and @Sneh1999 on this epic first Identity component.

File: src/components/OnchainName.test.tsx
Comment (cnasc) [inline src/components/OnchainName.test.tsx:30]: Does this need to be mocked? I notice that the implementation here is slightly different from the one in the real function. Is that significant?

File: src/components/OnchainName.tsx
Comment (cnasc) [reply src/components/OnchainName.tsx:26]: I think the idea here is to always display 
```suggestion
  const normalizedAddress = useMemo(() => {
```

Comment (cnasc) [reply src/components/OnchainName.tsx:26]: Also, I think the idea here is that the input address will display until ens name resolves. I don't think that's unreasonable, but maybe we'd prefer an explicit loading state?

File: src/store/inMemoryStorageService.ts
Comment (cnasc) [reply src/store/inMemoryStorageService.ts:5]: Do we see this as a frontend-only use case? If so, it might be best to expose this as a context

File: src/components/OnchainName.tsx
Comment (alvaroraminelli) [reply src/components/OnchainName.tsx:26]: Great call. Yes, we need loading state here. Wondering how the UX would look like when you have a list of users in the UX, will things shift and change like crazy. User might opt-in to display loading state. 

Need some UX thinking here.

File: src/store/inMemoryStorageService.ts
Comment (alvaroraminelli) [reply src/store/inMemoryStorageService.ts:5]: I see this evolving towards sharing in context as well. We might want build a light version of react-query for actions.

File: README.md
Comment (alvaroraminelli) [reply README.md:314]: soon we might have autogen docs that will be based on type correct?

File: README.md
Comment (alvaroraminelli) [reply README.md:311]: I am thinking add png file with little screenshot of the result? or we can have a link to stackbliz that show case the components.

File: src/components/OnchainName.tsx
Comment (alvaroraminelli) [reply src/components/OnchainName.tsx:7]: agreed with @cnasc. vote to accept to make less cumbersome for consumers.

File: src/store/client.ts
Comment (alvaroraminelli) [reply src/store/client.ts:4]: LGTM

File: package.json
Comment (alvaroraminelli) [inline package.json:21]: build fails if we remove the dependency

File: README.md
Comment (Zizzamia) [reply README.md:314]: Maybe... There is a difference between `type` information, and how to teach how to use something.

Vitepress will automate docs creation in some capacity, but the ART of teaching how to use a function is still a manual work.

File: README.md
Comment (Zizzamia) [reply README.md:311]: Let's circle back on this one. We still miss the default Tailwind css here.

File: src/store/inMemoryStorageService.ts
Comment (Zizzamia) [reply src/store/inMemoryStorageService.ts]: > I see this evolving towards sharing in context as well. 

React Context are a slippery slop, where folks think everything should be a context. Library should have like 90% of their state being just variable shared between functions. And for very very small exential stuff you can have Context that optimizes the Hooks story.

So before we build any React Context, it's important to ask the question, on where is the benefit for the end application. If it's just a nice to have for us as developer, than we don't need that. But again, for Hooks Rendering optimization, that we can talk about it.

File: package.json
Comment (Zizzamia) [reply package.json:21]: let's add it back.

**Comment by wespickett** [comment]
lgtm. We ended up switching to extend the next/jest config in build-onchain-apps, but if tests run successfully as is, then it should be fine with the current config https://github.com/coinbase/build-onchain-apps/blob/main/template/jest.config.js#L1

File: src/components/OnchainName.test.tsx
Comment (alvaroraminelli) [reply src/components/OnchainName.test.tsx]: tried the approach below but it is not working. fallback to mocking the entire function. 

```
import * as addressUtils from '../core/address';

beforeEach(() => {
    jest.clearAllMocks();
    jest.spyOn(addressUtils, 'getSlicedAddress');
  });

expect(addressUtils.getSlicedAddress).toHaveBeenCalledTimes(0);
```

File: src/components/OnchainName.tsx
Comment (Zizzamia) [inline src/components/OnchainName.tsx:35]: Not for later, this will cause a Layout Shift. Something to consider in follow up PR.

**Review by Zizzamia — APPROVED** [review]
BRILLIANT

**Pull request merged**
